emcc.py

```diff
@@ -667,19 +667,24 @@ def process_dynamic_libs(dylibs, lib_dirs):
 
   dylibs += extras
   for dylib in dylibs:
+    exports = webassembly.get_exports(dylib)
+    exports = set(e.name for e in exports)
+    settings.SIDE_MODULE_EXPORTS.extend(exports)
+
     imports = webassembly.get_imports(dylib)
     imports = [i.field for i in imports if i.kind in (webassembly.ExternType.FUNC, webassembly.ExternType.GLOBAL)]
-    settings.SIDE_MODULE_IMPORTS.extend(imports)
+    imports = set(i for i in imports if not i.startswith('invoke_'))
```

Comment:
a comment about why invoke is special would be good

(I assume because we autogenerate them at runtime?)

Reply:
Done.  Writing that comment made be realize the is future scope to improve this.